docs/application_migration.md

```diff
@@ -0,0 +1,16 @@
+## Checklist for migrating an Application to ArgoCD
+
+When migrating an application's deployment to be managed by ArgoCD use the following checklist to verify your process. 
+
+- [ ] Ensure your application manifests can be built using Kustomize.
+- [ ] If using secrets, make sure to include the .sops.yaml file in your repository.
+- [ ] Make sure to create the role granting access to namespace. See [here](cluster_ns_management.md#deploying-to-a-namespace) for more info.
```

Comment:
I think we should also track this role as part of the application manifests (and mention it here as well).

```suggestion
- [ ] Make sure to create the role granting access to namespace. See [here](cluster_ns_management.md#deploying-to-a-namespace) for more info. This role is tracked in your application manifests.
```